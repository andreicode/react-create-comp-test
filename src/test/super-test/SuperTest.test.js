import React from 'react';
import { render } from '@testing-library/react';
import SuperTest from './SuperTest';

describe('[COMPONENT] SuperTest', () => {
  it('renders correctly', () => {
    const { container } = render(<SuperTest />);
    expect(container).toMatchSnapshot();
  });

  it('contains "SuperTest" text', () => {
    const { getByText } = render(<SuperTest />);
    const textElement = getByText(/SuperTest/i);
    expect(textElement).toBeInTheDocument();
  });

  it('contains "SuperTest" text', () => {
    const { getByText, container } = render(<SuperTest />);
    const btnElement = getByText(/PressMe/i);
    expect(btnElement).toBeInTheDocument();
    const counter = container.querySelector('#count');
    expect(counter.textContent).toBe('0');
    btnElement.click();
    expect(counter.textContent).toBe('1');
  });
});
