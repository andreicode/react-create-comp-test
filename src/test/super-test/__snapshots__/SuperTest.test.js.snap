// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[COMPONENT] SuperTest renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <p>
      SuperTest
    </p>
    <p
      id="count"
    >
      0
    </p>
    <button>
      PressMe
    </button>
  </div>
</div>
`;
